vm/parser.ts

```diff
@@ -1070,6 +1070,19 @@ module J2ME {
     return cache[hash & 0xff];
   }
 
+  /**
+   * Generate an ID for a Class.  We use this to map Java objects to their
+   * ClassInfo objects, storing the ID for the Class in the first four bytes
+   * of the memory allocated for the Java object in the ASM heap.
+   *
+   */
+  export var generateClassId = (function() {
+    var nextId = 1;
+    return function() {
+      return nextId++;
+    }
+  })();
```

Comment:
This doesn't appear to be used outside ClassInfo, so it'd be better as a private static property of that class.